cmd/timerdisplay.go

```diff
@@ -0,0 +1,143 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gosuri/uilive"
+)
+
+// TimerDisplay displays timing for incremental updates, like plan and apply
+type TimerDisplay struct {
+	writer  *uilive.Writer
+	once    sync.Once
+	started bool
+	logger  *logrus.Entry
+
+	items []*timerDisplayTimer
+	wlock *sync.RWMutex
+}
+
+func (td *TimerDisplay) init() {
+	td.writer = uilive.New()
+	td.writer.RefreshInterval = 250 * time.Millisecond
+	td.writer.Out = os.Stderr
+
+	td.logger = logrus.WithField("component", "timer display")
+	td.logger.Logger.Out = td.writer.Bypass()
+
+	td.wlock = new(sync.RWMutex)
+}
+
+// Message creates the current message to display
+func (td *TimerDisplay) Message() string {
+	td.once.Do(td.init)
+
+	out := "\nActive:\n"
```

Comment:
I noticed this output is sometimes displayed on one line and sometimes displayed twice, with the second a few lines below. The second instance flashes on and off the screen.